### ISSUE TITLE
PHQL Subquery with JOINS duplicates subquery join

### ISSUE THREAD

**Issue by Surt** [phalcon/cphalcon#10505]
I'm having troubles with 2.0.3 in subqueries when I use a join on the subquery and the main query. It seems that phalcon duplicates the join, giving exceptions due to duplicate alias.

An example (not actual usage but tested)

```php
        $phql = '
             SELECT * FROM App\User\Models\User user
             JOIN App\User\Models\UserFriendship as friendship ON friendship.user_id = user.id
             WHERE user.id NOT IN (
              SELECT user.id FROM App\User\Models\User user
              JOIN App\User\Models\UserFriendship as friendship ON friendship.user_id = user.id
             )';

        $this->modelsManager->createQuery($phql)->execute();
``` 

the warning shows the sql made:

``` 
    Phalcon\Db\Adapter\Pdo->query(
SELECT `user`.`id` AS `_user_id`, `user`.`name` AS `_user_name`, `user`.`email` AS `_user_email`, `user`.`telephone` AS `_user_telephone`, `user`.`password` AS `_user_password`, `user`.`avatar` AS `_user_avatar`, `user`.`gender` AS `_user_gender`, `user`.`birth` AS `_user_birth`, `user`.`role` AS `_user_role`, `user`.`active` AS `_user_active`, `user`.`date_created` AS `_user_date_created`, `user`.`date_updated` AS `_user_date_updated`, `user`.`deleted` AS `_user_deleted`, `friendship`.`id` AS `_friendship_id`, `friendship`.`user_id` AS `_friendship_user_id`, `friendship`.`friend_id` AS `_friendship_friend_id`, `friendship`.`status` AS `_friendship_status` 
FROM `user` AS `user` 
INNER JOIN `user_friendship` AS `friendship` ON `friendship`.`user_id` = `user`.`id`
 WHERE `user`.`id` NOT IN (
     SELECT `user`.`id` AS `id`
 FROM `user` AS `user` 
INNER JOIN `user_friendship` AS `friendship` ON `friendship`.`user_id` = `user`.`id` 
INNER JOIN `user_friendship` AS `friendship` ON `friendship`.`user_id` = `user`.`id`)
, null, null)

``` 
It does not matter the tables (in this example those are the same because I took the first one I saw, but even with different joined tables the subquery join is repeated).

during the weekend I will try to post here the full code for test.

**Comment by dschissler** [comment]
I'm also getting an additional JOIN in another case.  One of implicitly added from the hasManyToMany relationship and then when the intermediate table is explicitly joined.

```php
    $Persons = "Webird\\Models\\Persons";
    $Groups = "Webird\\Models\\Groups";
    $PersonsGroups = "Webird\\Models\\PersonsGroups";
    $phql = "SELECT $Persons.* FROM $Persons JOIN $PersonsGroups JOIN $Groups WHERE $Groups.id = :id: AND $PersonsGroups.type = :type: LIMIT 1";
```

The workaround is to remove `JOIN $PersonsGroups` so that only the implicit JOIN is used.

**Comment by hakimio** [comment]
Any update on this?
I think it should be "high" level bug instead of "medium". Not being able to use joins in subqueries is pretty big deal...

**Comment by andresgutierrez** [comment]
@hakimio Only security bugs are marked as high

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
